### PR TITLE
Updated warnings for anomaly decection including language determined on entity

### PR DESCRIPTION
**Pull request by eric-j**

### CODE REVIEW

File: shared/agent/src/providers/newrelic.ts [hunk 1]
```diff
@@ -961,35 +963,35 @@ export class NewRelicProvider
...
+							languageAndVersionValidation: languageAndVersionValidation,
+						} as EntityAccount;
+					})
+				);
+				mappedUniqueEntities = mappedUniqueEntities.filter(Boolean);
+				mappedUniqueEntities.sort((a, b) =>
+					`${a?.accountName}-${a?.entityName}`.localeCompare(`${b?.accountName}-${b?.entityName}`)
+				);
 				response.repos?.push({
 					repoId: repo.id!,
```
Comment (dsellarsnr) [inline shared/agent/src/providers/newrelic.ts:989]: can remove the ! - you already have a guard

File: shared/agent/src/providers/newrelic.ts [hunk 2]
```diff
@@ -1020,7 +1022,7 @@ export class NewRelicProvider
 		return tracingValue === "standard" || tracingValue === "infinite";
 	}
 
-	private languageAndVersionValidation(entity?: Entity): object {
+	private async languageAndVersionValidation(entity?: Entity): Promise<object> {
```
Comment (dsellarsnr) [inline shared/agent/src/providers/newrelic.ts:1025]: Give it the full LanguageAndVersionValidation type

**Review by dsellarsnr — APPROVED** [review]
Just 2 smol things

File: shared/agent/src/providers/newrelic.ts [hunk 3]
```diff
@@ -1020,7 +1022,7 @@ export class NewRelicProvider
...
 		const tags = entity?.tags || [];
 		const agentVersion = tags.find(tag => tag.key === "agentVersion");
```
Comment (marcelofarias) [inline shared/agent/src/providers/newrelic.ts:1027]: Remind me to create some accessor methods in Entity to return those values. I've implemented the same logic somewhere else.

**Pull request merged**